core/src/test/java/org/fao/geonet/geocat/kernel/AapMetadataReportTest.java

```diff
@@ -0,0 +1,127 @@
+        assertTrue("Unexpected geodata ID",
+                el.contains("<identifier>101.2-TG</identifier>"));
+        assertTrue("Unexpected UUID",
+                el.contains("<uuid>d2ab7e4e-d135-4442-b0af-2f8892d87843</uuid>"));
+        assertTrue("Unexpected geodatatype",
+                el.contains("<geodatatype>Ja</geodatatype>"));
+        assertTrue("Unexpected owner",
+                el.contains("<owner>SFF</owner>"));
+        assertTrue("Unexpected owner",
+        el.contains("<specialistAuthority>Camptocamp</specialistAuthority>"));
+        assertTrue("Unexpected update frequency",
+                   el.contains("<updateFrequency>userDefined</updateFrequency>"));
+        assertTrue("Unexpected duration of conservation",
+                   el.contains("<durationOfConservation>3</durationOfConservation>"));
+        assertTrue("Unexpected comment on duration",
+                   el.contains("<commentOnDuration>Sample comment on the duration of conservation</commentOnDuration>"));
+        assertTrue("Unexpected comment on archival",
+                   el.contains("<commentOnArchival>Sample comment on "
+                           + "the archival value</commentOnArchival>"));
+        assertTrue("Unexpected appraisal of archival value",
+                   el.contains("<appraisalOfArchival>S</appraisalOfArchival>"));
+        assertTrue("Unexpected reason for archiving value",
+                   el.contains("<reasonForArchiving>definingPowers</reasonForArchiving>")
+                );
+    }
+
+    @Test
+    public void extractAapInfoOnNonAapMdTest() throws Exception {
+        URL rawMdUrl = this.getClass().getResource("mdaaptest-noaap.xml");
+        assumeTrue(rawMdUrl != null);
+        File rawMdF = new File(rawMdUrl.toURI());
+        assumeTrue(rawMdF.exists());
+
+        String rawMd = FileUtils.readFileToString(rawMdF);
+        Metadata tested = new Metadata();
+        tested.setData(rawMd);
+
+        String el = Xml.getString(amr.extractAapInfo(tested));
+
+        assertTrue("Unexpected content element extracted from mdaaptest-noaap.xml",
+                el.contains("<durationOfConservation />") &&
+                el.contains("<commentOnDuration />")      &&
+                el.contains("<commentOnArchival />")      &&
+                el.contains("<reasonForArchiving />")     &&
+                el.contains("<appraisalOfArchival />"));
+    }
+    
+    @Test
+    public void extractAapInfoFromCustomerProvidedMd() throws Exception {
+        URL rawMdUrl = this.getClass().getResource("aap.xml");
+        assumeTrue(rawMdUrl != null);
+        File rawMdF = new File(rawMdUrl.toURI());
+        assumeTrue(rawMdF.exists());
+        
+
+        String rawMd = FileUtils.readFileToString(rawMdF);
+        Metadata tested = new Metadata();
+        tested.setData(rawMd);
+
+        
+        String el = Xml.getString(amr.extractAapInfo(tested));  
+        System.out.println(el);
```

Comment:
maybe we can test string length like `AssertTrue(el.length() > 10)`.
If test fails then an exception is throw ?